common/js/theming.js

```diff
@@ -0,0 +1,226 @@
+		'desc': 'The size of the font used in the player.',
+		'property': 'font-size',
+		'func': function(a) {
+					return a;
+				}
+	},
+	{
+		'name': 'Text Color',
+		'desc': 'The color of the text in the player.',
+		'type': 'color',
+		'property': 'text-color',
+		'func': function(a) { return a; }
+	},
+	{
+		'name': 'Secondary Text Color',
+		'desc': 'The color of the text used in low-contrast areas.',
+		'type': 'color',
+		'property': 'inverse-color',
+		'func': function(a) { return a; }
+	},
+	{
+		'name': 'Accent Color',
+		'desc': 'The color used in accents in the player.',
+		'type': 'color',
+		'property': 'accent-color',
+		'func': function(a) { return a; }
+	},
+	{
+		'name': 'Focused Input Accent Color',
+		'desc': 'The color of borders of focused inputs.',
+		'type': 'color',
+		'property': 'accent-color-darker',
+		'func': function(a) { return a; }
+	},
+	{
+		'name': 'Tab Size',
+		'desc': 'The width of the buttons used to switch which panel you\'re on.',
+		'property': 'tabSize',
+		'func': function(a) {
+					return a;
+				}
+	},
+	{
+		'name': 'Warning Background',
+		'desc': 'The background used for warnings in the player.',
+		'type': 'color',
+		'property': 'warning-bgcolor',
+		'func': function(a) {
+			return a;
+		}
+	},
+	{
+		'name': 'Album Background',
+		'desc': 'The background for albums that use the default cover image.',
+		'type': 'color',
+		'property': 'album-bg',
+		'func': function(a) {return a;}
+	}
+]
+
+function export2json() {
+  let data = JSON.parse(JSON.stringify(themeInfo)) // dereference
+  data["A comment to power-users"] = "You can set backgrounds to an image by setting 'background' or 'album-bg' to \"url('/path-to-image')\".";	
+  const a = document.createElement("a");
+  a.href = URL.createObjectURL(new Blob([JSON.stringify(data, null, 2)], {
+    type: "application/json"
+  }));
+  a.setAttribute("download", "theme.json");
+  document.body.appendChild(a);
+  a.click();
+  document.body.removeChild(a);
+}
+document.getElementById('inport').addEventListener('change', handleFileSelect, false);
+document.getElementById('export').addEventListener('click', export2json);
+
+function handleFileSelect(event){
+  const reader = new FileReader()
+  reader.onload = handleFileLoad;
+  reader.readAsText(event.target.files[0])
+}
+function handleFileLoad(event){
+	//verify file
+	let res = event.target.result;
+	try {
+		JSON.parse(res);
+	} catch(e) {
+		alert('Invalid file: '+e);
+		return;
+	}
+	let resParsed = JSON.parse(res);
+	for (let key in themeInfo) {
+		if (!resParsed.hasOwnProperty(key)) {
+			alert('File is missing property: '+key);
+			return;
+		}
+		if (themeInfo[key].type && themeInfo[key].type == "color" && !(/^#([0-9A-F]{3}){1,2}$/i.test(resParsed[key]))) {
+			alert(`${key}: Please use hex color codes, not '${resParsed[key]}'`);
+			return;
+		}
+	}
+	// do stuff
+	for (let key in themeInfo) {
+		setVar(key, resParsed[key]);
+		controls[key].input.value = resParsed[key];
```

Comment:
I think this block of code needs
`themeInfo[key] = resParsed[key];`

without this I wasn't able to load any themes from files

Reply:
Absolutely right. Whoopsie. I'll add that in.